packages/core/src/main.ts

```diff
@@ -0,0 +1,144 @@
+    options?: GetWalletOptions,
+  ) => Promise<StarknetWindowObject[]> // Returns only preauthorized wallets available in the window object
+  getDiscoveryWallets: (options?: GetWalletOptions) => Promise<WalletProvider[]> // Returns all wallets in existence (from discovery file)
+  getLastConnectedWallet: () => Promise<StarknetWindowObject | null> // Returns the last wallet connected when it's still connected
+  getDefaultWallet: () => Promise<StarknetWindowObject | null> // Returns the default wallet
+  enable: (
+    wallet: StarknetWindowObject,
+    options?: {
+      starknetVersion?: "v3" | "v4"
+    },
+  ) => Promise<ConnectedStarknetWindowObject> // Connects to a wallet
+  disconnect: (options?: { clearDefaultWallet?: boolean }) => Promise<void> // Disconnects from a wallet
+}
+
+export function getSnWallet(
+  options: Partial<SnWalletOptions> = {},
+): SnWalletResult {
+  const { storageFactoryImplementation, windowObject, isWalletObject } = {
+    ...defaultOptions,
+    ...options,
+  }
+  const lastConnectedStore = storageFactoryImplementation("gsw-last")
+  const defaultWalletStore = storageFactoryImplementation("gsw-default")
+
+  return {
+    getAvailableWallets: async (options = {}) => {
+      const availableWallets = scanObjectForWallets(
+        windowObject,
+        isWalletObject,
+      )
+      return pipe<StarknetWindowObject[]>(
+        (_) => filterBy(_, options),
+        (_) => sortBy(_, options.sort),
+      )(availableWallets)
+    },
+    getPreAuthorizedWallets: async (options = {}) => {
+      const availableWallets = scanObjectForWallets(
+        windowObject,
+        isWalletObject,
+      )
+      return pipe<StarknetWindowObject[]>(
+        (_) => filterByPreAuthorized(_),
+        (_) => filterBy(_, options),
+        (_) => sortBy(_, options.sort),
+      )(availableWallets)
+    },
+    getDiscoveryWallets: async (options = {}) => {
+      return pipe<WalletProvider[]>(
+        (_) => filterBy(_, options),
+        (_) => sortBy(_, options.sort),
+      )(discovery)
+    },
+    getLastConnectedWallet: async () => {
+      const lastConnectedWalletId = lastConnectedStore.get()
+      const allWallets = scanObjectForWallets(windowObject, isWalletObject)
+      const lastConnectedWallet = allWallets.find(
+        (w) => w.id === lastConnectedWalletId,
+      )
+      const [firstPreAuthorizedWallet] = await filterByPreAuthorized(
+        lastConnectedWallet ? [lastConnectedWallet] : [],
+      )
+
+      return firstPreAuthorizedWallet ?? null
+    },
+    getDefaultWallet: async () => {
+      const defaultWalletId = defaultWalletStore.get()
+      const allWallets = scanObjectForWallets(windowObject, isWalletObject)
+      const defaultWallet = allWallets.find((w) => w.id === defaultWalletId)
+      const [firstPreAuthorizedWallet] = await filterByPreAuthorized(
+        defaultWallet ? [defaultWallet] : [],
+      )
+
+      return firstPreAuthorizedWallet ?? null
```

Comment:
need to clear `defaultWalletStore ` value when `firstPreAuthorizedWallet` is nullish